core/test/matrix/CMakeLists.txt

```diff
@@ -5,6 +5,15 @@ ginkgo_create_test(csr_builder)
 ginkgo_create_test(dense)
 ginkgo_create_test(diagonal)
 ginkgo_create_test(ell)
+add_library(test_fbcsr_sample fbcsr_sample.cpp)
+target_include_directories(test_fbcsr_sample
+  PRIVATE
+  "$<BUILD_INTERFACE:${Ginkgo_SOURCE_DIR}/include>"
+  "$<BUILD_INTERFACE:${Ginkgo_SOURCE_DIR}>"
+  "$<BUILD_INTERFACE:${Ginkgo_BINARY_DIR}/include>"
+  )
+ginkgo_create_test(fbcsr test_fbcsr_sample)
```

Comment:
I do not go into this pr yet. but some comments maybe help the windows issue
does it build another library (*.so in Linux)? Moreover, do you also use it for reference test?
Maybe you need to cp/mv the library to the shared library path in shared folder, which we did in `ginkgo_compile_features`.
The error message is the same in static and shared build. (static complains complex only, shared complains all valuetype)
For shared build, it could be related to the above issue, but it should not help static build.
Static build only complains the complex of fbcsr, does it lose the complex instantiate?